drivers/include/opt3001.h

```diff
@@ -0,0 +1,152 @@
+ *     \f{eqnarray*}{
+ *         Lux &=& LSB\_Size \cdot R[11:0]
+ *     \f}
+ * Where LSB\_Size:
+ *     \f{eqnarray*}{
+ *         LSB\_Size &=& 0.01 \cdot 2^{E[3:0]}
+ *     \f}
+ *
+ * The calculation is wrapped from TI OPT3001 Ambient Light Sensor (ALS) datasheet (Rev. C).
+ *
+ * This driver provides @ref drivers_saul capabilities.
+ * @{
+ *
+ * @file
+ * @brief       Interface definition for the OPT3001 sensor driver.
+ *
+ * @author      Jannes Volkens <jannes.volkens@haw-hamburg.de>
+ */
+
+#ifndef OPT3001_H
+#define OPT3001_H
+
+#include <stdint.h>
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   OPT3001 Default Address
+ *
+ * If set to 0x44 the corresponding ADDR PIN is GND.
+ */
+#ifndef OPT3001_I2C_ADDRESS
+#define OPT3001_I2C_ADDRESS   (0x45)
+#endif
+
+/**
+ * @name    Conversion time
+ * @{
+ */
+#define OPT3001_CONVERSION_TIME_100_MS   (0x0000) /**< Conversion time of 100ms */
+#define OPT3001_CONVERSION_TIME_800_MS   (0x0800) /**< Conversion time of 800ms */
+/** @} */
+
+/**
+ * @brief   OPT3001 Default conversion time
+ *
+ * If set to 0x0000, the conversion time will be 100ms.
+ * If set to 0x0800, the conversion time will be 800ms
+ */
+#ifndef OPT3001_CONVERSION_TIME
+#define OPT3001_CONVERSION_TIME   OPT3001_CONVERSION_TIME_800_MS
+#endif
+
+/**
+ * @brief   Parameters needed for device initialization
+ */
+typedef struct {
+    i2c_t i2c_dev; /**< I2C device, the sensor is connected to */
+    uint8_t i2c_addr; /**< The sensor's slave address on the I2C bus */
+} opt3001_params_t;
+
+/**
+ * @brief   Device descriptor for OPT3001 sensors
+ */
+typedef struct {
+    opt3001_params_t params; /**< Configuration parameters */
+} opt3001_t;
+
+/**
+ * @brief   OPT3001 specific return values
+ */
+enum {
+    OPT3001_OK,         /**< Success, no error */
+    OPT3001_ERROR_BUS,  /**< I2C bus error */
+    OPT3001_ERROR_DEV,  /**< Internal device error */
+    OPT3001_ERROR       /**< General error */
+};
+
+/**
+ * @brief   Initialize the OPT3001 sensor driver.
+ *
+ * @param[out] dev          device descriptor of sensor to initialize
+ * @param[in]  params       configuration parameters
+ *
+ * @return                  0 on success
+ * @return                  -OPT3001_ERROR_BUS on I2C bus error
+ * @return                  -OPT3001_ERROR_DEV if sensor test failed
+ */
+int opt3001_init(opt3001_t *dev, const opt3001_params_t *params);
+
+/**
+ * @brief   Reset the OPT3001 sensor, afterwards it should be reinitialized.
+ *
+ * @param[out] dev          device descriptor of sensor
+ *
+ * @return                  0 on success
+ * @return                  -1 on error
+ */
+int opt3001_reset(const opt3001_t *dev);
+
+/**
+ * @brief   Set active mode, this enables periodic measurements.
+ *
+ * @param[in]  dev          device descriptor of sensor
+ *
+ * @return                  0 on success
+ * @return                  -1 on error
+ */
+int opt3001_set_active(const opt3001_t *dev);
```

Comment:
You have a function to activate the sensor, but none to deactivate it again.
Or is the idea to simply call `opt3001_reset()` to put the sensor back to idle? If so, please add this to the function documentation.